lightningd/gossip_control.c

```diff
@@ -296,7 +296,9 @@ static void json_getroute_reply(struct subd *gossip, const u8 *reply, const int
 static void json_getroute(struct command *cmd, const char *buffer, const jsmntok_t *params)
 {
 	struct pubkey id;
-	jsmntok_t *idtok, *msatoshitok, *riskfactortok, *cltvtok;
+	struct pubkey from_id;
+	struct pubkey *from_id_to_use;
```

Comment:
`from_id_to_use` is redundant, we can just initialize `from_id` with our own id, and overwrite if necessary.